### ISSUE TITLE
Potential unauthorized access to underlying database

### ISSUE THREAD

**Issue by politician** [influxdb/influxdb#1364]
I was looking through the HTTP handler code, and noticed that access to the `/metastore` route is not protected in the same way that access to the `/query` or `/write` endpoints are.  Specifically, the `/metastore` [handler](https://github.com/influxdb/influxdb/blob/master/httpd/handler.go#L211) would not be protected with an `authenticate` middleware because the method signature [does not opt-in](https://github.com/influxdb/influxdb/blob/master/httpd/handler.go#L87-L90) to authentication.

```Go
// If it's a handler func that requires authorization, wrap it in authorization
if hf, ok := r.handlerFunc.(func(http.ResponseWriter, *http.Request, *influxdb.User)); ok {
	handler = authenticate(hf, h, requireAuthentication)
}
```

This unprotected pathway results in a call to BoltDB's [Tx.Copy](https://github.com/boltdb/bolt/blob/master/tx.go#L254-L258) method which apparently copies the entire database to the given writer.

```Go
// Copy writes the entire database to a writer.
// A reader transaction is maintained during the copy so it is safe to continue
// using the database while a copy is in progress.
// Copy will write exactly tx.Size() bytes into the writer.
func (tx *Tx) Copy(w io.Writer) error { /* ... */ }
```

So, bottom line, this is probably a bug.  Personally, I'd want this operation to be restricted to cluster admins and other authenticated nodes (it looks like this capability is used as part of the clustering work for 0.9.0?)

**Comment by otoolep** [comment]
Thanks @politician -- yes, this is part of the 0.9.0 clustering, and the 0.9.0 code is still very much in progress. We'll get around to securing the routes you mention soon. The are somewhat different since they are not meant to be accessed directly by users.

**Comment by politician** [comment]
Great, thanks.

**Comment by beckettsean** [comment]
`/metastore` responds with 404, I assume it no longer exists. Was likely ripped out when raft was redone.
